studio/components/interfaces/SQLEditor/UtilityPanel/UtilityActions.tsx

```diff
@@ -23,10 +26,31 @@ const UtilityActions = ({
   executeQuery,
 }: UtilityActionsProps) => {
   const os = detectOS()
+  const snap = useSqlEditorStateSnapshot()
+  const [isDownloadSnippetModalOpen, setIsDownloadSnippetModalOpen] = useState(false)
+
+  const snippet = snap.snippets[id]?.snippet
 
   return (
     <>
+      {snippet && (
+        <DownloadSnippetModal
+          id={id}
+          visible={isDownloadSnippetModalOpen}
+          onCancel={() => setIsDownloadSnippetModalOpen(false)}
+        />
+      )}
       <SavingIndicator id={id} />
+      {IS_PLATFORM && snippet && (
+        <Button
+          type="text"
+          size="tiny"
+          icon={<IconDownload size="tiny" />}
+          onClick={() => setIsDownloadSnippetModalOpen(true)}
+        >
+          Download as migration file
```

Comment:
i'd opt to put "Download as migration file" behind a tooltip to prevent clutter here

same considerations for table editor header - lets keep things neat and clean while we still can

Reply:
Can i also check if this is a heavily requested feature? Alternatively we put this in the snippet's dropdown in the left side nav

Reply:
Good points @joshenlim - how about we put it under the snippet's left side dropdown for now and we can make it more visible in the future if there's lots of usage